src/lxd.py

```diff
@@ -47,7 +47,22 @@ def mk_dir(self, dir_name: str) -> None:
         Args:
             dir: Name of the directory to create.
         """
-        self.instance.execute(["/usr/bin/mkdir", "-p", dir_name])
+        lxc_command = [
+            "/snap/bin/lxc",
+            "exec",
+            self.instance.name,
+            "--",
+            "/usr/bin/mkdir",
+            "-p",
+            dir_name,
+        ]
+        try:
+            execute_command(lxc_command)
+        except SubprocessError as err:
+            logger.exception("Failed to create directory")
+            raise LxdError(
+                f"Unable to create directory in LXD instance {self.instance.name}"
+            ) from err
```

Comment:
Why making this change ?

Reply:
I think this is the residue of an early debug attempt, removed, thanks!